### PR TITLE
add optional selinux support

### PR DESCRIPTION
**Pull request by jlambert121**

### CODE REVIEW

**Comment by jlambert121** [comment]
The build failures look like they exist on master as well.

File: manifests/init.pp [hunk 1]
```diff
@@ -737,6 +745,14 @@
     }
   }
 
+  if $::selinux and $manage_selinux_real and $sshd_config_port != '22' {
```
Comment (ghoneycutt) [inline manifests/init.pp]: Why would we not do this if the port is the default 22/tcp ?

Comment (jlambert121) [inline manifests/init.pp]: SELinux comes with that label by default so just avoiding the exec on the machine if it's still on port 22.  I could see an argument to just always include it too though - either way.

**Comment by ghoneycutt** [comment]
PR #162 is adding the ability to have multiple ports, so this would need to work with that approach.

Could we use the selmodule type to handle this instead of an exec? I don't think that code around handling selinux should be in the ssh module. I would rather it be elsewhere and the ssh module be able to use it, similar to how the firewall rules are handled.

https://docs.puppet.com/puppet/3.8/reference/types/selmodule.html

**Comment by jlambert121** [comment]
Multiple ports - I'll have to look into that some more.  You want to maintain puppet 3.x compatibility, correct?

I don't think the selmodule type will really work here, but maybe someone can give me more info on how it would.  I considered adding [a selinux](https://forge.puppetlabs.com/jfryman/selinux) module, but was trying to avoid that dependency.  SELinux support in puppet isn't awesome - I've thought adding some native types/providers but just haven't gotten around to it.

File: manifests/init.pp (hunk 1 above)
Comment (ghoneycutt) [inline manifests/init.pp]: We don't want to make assumptions on the initial state of the system. It is better to be declarative and say that we want it and the idempotent nature of Puppet will ensure it is there.

**Comment by ghoneycutt** [comment]
Yes, puppet v3 compatibility is required. Perhaps there is another selinux module with types/providers that would work? I'm ok with having a dependency on that as it would be the same as the firewall dependency.

**Comment by ghoneycutt** [comment]
Closing due to age. The module has undergone a major release and now supports every documented configuration parameter for OpenSSH. The module also allows for custom entries to future proof functionality and support other SSH implementations. See [UPGRADING.md](https://github.com/ghoneycutt/puppet-module-ssh/blob/main/UPGRADING.md).

**Pull request closed without merging**